### PR TITLE
fix(compiler): ignore @import in comments

### LINKED ISSUE #12196: The styles parser handles @import statements in comments and tries to load invalid resources
**I'm submitting a ...**  (check one with "x")

```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

A `styles: ['/*@import "foo!"*/']` results in 

```
Unhandled Promise rejection: Failed to load foo! ; Zone: <root> ; Task: Promise.then ; Value: Failed to load foo! undefined
zone.js (355,13)
```

**Expected behavior**

This shouldn't try to load
**Minimal reproduction of the problem with instructions**

http://plnkr.co/edit/Y2bcV71JNJbSyoteBbez?p=preview
**What is the motivation / use case for changing the behavior?**

So it actually works
**Please tell us about your environment:**

- **Angular version:** 2.0.2
  
  
- **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
  
  
  
  Funny, still no Edge...
- **Language:** [all | TypeScript X.X | ES6/7 | ES5]
- **Node (for AoT issues):** `node --version` =

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Fixes https://github.com/angular/angular/issues/12196

### CODE REVIEW

File: modules/@angular/compiler/test/style_url_resolver_spec.ts [hunk 1]
```diff
@@ -36,6 +36,13 @@ export function main() {
       expect(styleWithImports.styleUrls).toEqual(['http://ng.io/1.css', 'http://ng.io/2.css']);
     });
 
+    it('should ignore "@import" in comments', () => {
+      const css = '/*@import "foo!"*/';
+      const styleWithImports = extractStyleUrls(urlResolver, 'http://ng.io', css);
+      expect(styleWithImports.style.trim()).toEqual('');
```
Comment (vicb) [inline modules/@angular/compiler/test/style_url_resolver_spec.ts]: I think the test should be:
```ts
expect(...).toContain(nonCommentedImport);
expect(...).not.toContain(commentedImport);
```

Also please you PR two commits:
- refactor,
- fix.
Thanks

File: modules/@angular/compiler/src/style_url_resolver.ts [hunk 2]
```diff
@@ -30,7 +28,11 @@ export function isStyleUrlResolvable(url: string): boolean {
 export function extractStyleUrls(
     resolver: UrlResolver, baseUrl: string, cssText: string): StyleWithImports {
   const foundUrls: string[] = [];
-  const modifiedCssText = cssText.replace(_cssImportRe, function(...m: string[]) {
+
+  // remove comments
+  cssText = cssText.replace(CSS_COMMENT_REGEXP, '');
+
+  const modifiedCssText = cssText.replace(CSS_IMPORT_REGEXP, (...m: string[]) => {
```
Comment (vicb) [inline modules/@angular/compiler/src/style_url_resolver.ts]: you can also chain replace

Comment (DzmitryShylovich) [inline modules/@angular/compiler/src/style_url_resolver.ts]: http://prntscr.com/dhvn3r
this is how it looks likes chained.

**Comment by DzmitryShylovich** [comment]
done

**Comment by DzmitryShylovich** [comment]
@vicb ping

**Comment by vicb** [comment]
Thanks !

**Pull request merged**